### PR TITLE
fix: Stringop overread warning in GCC12

### PR DESCRIPTION
**Pull request by paulgessinger**
This showed up in our GCC12 based LCG builds, and only in Release mode. I don't think there's a way for us to fix this because the warning actually comes from `std::move`'ing `boost::small_vectors`. I'm suppressing the warnings here and remove the job from being allowed to fail.

### CODE REVIEW

File: Core/include/Acts/Seeding/BinnedSPGroup.ipp [hunk 1]
```diff
@@ -74,7 +74,18 @@ Acts::BinnedSPGroupIterator<external_spacepoint_t>::operator*() const {
           m_group->m_bins[m_current_localBins[INDEX::Z]],
           m_group->m_grid.get());
 
+  // GCC12+ in Release throws an overread warning here due to the move.
+  // This is from inside boost code, so best we can do is to suppress it.
+#if defined(__GNUC__) && !defined(__clang__)
```
Comment (benjaminhuth) [inline Core/include/Acts/Seeding/BinnedSPGroup.ipp:79]: why not

```suggestion
#if defined(__GNUC__) && __GNUC__ >= 11 && !defined(__clang__)
```
and omitting the second `#if`?

Comment (paulgessinger) [reply Core/include/Acts/Seeding/BinnedSPGroup.ipp:79]: Can do!

Comment (paulgessinger) [reply Core/include/Acts/Seeding/BinnedSPGroup.ipp]: Done now.

**Review by benjaminhuth — APPROVED** [review]
Looks good to me!

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107653) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107652) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107651) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107650) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107649) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107648) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107647) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107646) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107645) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107644) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107643) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107642) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107641) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107640) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107639) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107638) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32107637) |

**Pull request merged**